rails/daylight/refiners.rb

```diff
@@ -126,7 +127,15 @@ def reflection_names
     # Supplies where conditions and returns the resulting +ActiveRecord::Relation+.
     # Raises +ArgumentError+ if the keys are not valid attributes on the model.
     def filter_by params
-      where (params||{}).with_indifferent_access.assert_valid_keys(attribute_names + reflection_names)
+      params = params || {}
+      dummy = self.new(params)
+      params = params.with_indifferent_access rescue params.to_h
+      params.each do |k, v|
+        if dummy[k].class == TrueClass || dummy[k].class == FalseClass
```

Comment:
why would you check for false?

Reply:
This one was pretty painful and this block of code is a workaround to that issue. 
When working from GFE I had the issue were daylight was not returning the right stuff for queries such as. Merchant.where(active: "false") or Merchant.where(active: "1"). Effectively, anything that wasn't 0 or 1 evaluated to false for purposes of the AR query.

Reply:
but you're only mutating params if they are `TrueClass` or `FalseClass`, and in both of those cases, it'll end up the same.

Reply:
I'm checking the "booleanity" of the model's attributes. If it's a boolean model I need to cast the attribute value into a boolean which normally happens automagically afaik but these issue was persistent so this manual casting was my fix.

Reply:
but you're not casting anything -- if it's `1`, then it wont match on `== TrueClass`, if the param is `true`, then you cast it to `true`. In the end, nothing changes.